Homeworks/Homework_3/script.py

```diff
@@ -0,0 +1,369 @@
+    :param csv_file_path: str, path to the folder where the csv file is located
+    """
+    with open(csv_file_path, 'w', encoding='utf-8') as csv_file:
+        csv_file.write(csv_data)
+
+
+def read_data_from_csv_file(csv_file_path):
+    """
+    Read data from csv file and write it to variable using csv lib
+
+    :param csv_file_path: str, path to the folder where the csv file is located
+    :return: list, data that was read from the csv file
+    """
+    with open(csv_file_path, 'r', encoding='utf-8') as csv_file:
+        return [row for row in csv.DictReader(csv_file)]
+
+
+def filter_data(csv_data, filtering_gender, filtering_number_of_rows):
+    """
+    Filter data using match cases
+
+    :param csv_data: str, data which has to be filtered
+    :param filtering_gender: str, a gender by which the data has to be filtered
+    :param filtering_number_of_rows: int, a number of rows to be filtered
+    :return: list, filtered data
+    """
+    if filtering_gender is not None:
+        return list(filter(lambda row: row['gender'] == filtering_gender, csv_data))
+    elif filtering_number_of_rows is not None:
+        return csv_data[:filtering_number_of_rows]
+    else:
+        return csv_data
+
+
+def rearrange_datetime_data(user_datetime_data, string_format):
+    """
+    Reformat the value of a user field to another format
+
+    :param user_datetime_data: str, the value of the field to be reformatted
+    :param string_format: str, preferred format of value
+    :return:
+    """
+    return datetime.strptime(user_datetime_data, '%Y-%m-%dT%H:%M:%S.%fZ').strftime(string_format)
+
+
+def rearrange_name_data(user_name_title):
+    """
+    Change form of the user's name prefix based on their current value
+
+    :param user_name_title: str, current form of the name prefix
+    :return: str, changed form of the name prefix
+    """
+    match user_name_title:
+        case 'Mrs':
+            return 'missis'
+        case 'Ms':
+            return 'miss'
+        case 'Mr':
+            return 'mister'
+        case 'Madame':
+            return 'mademoiselle'
+
+
+def rearrange_user_time_data(user_timezone_offset):
+    """
+    Calculates the user's time based on their time zone
+
+    :param user_timezone_offset: str, current user timezone offset
+    :return: str, current user time
+    """
+    user_timezone = user_timezone_offset.split(':')
+    hours = int(user_timezone[0][1:]) \
+        if user_timezone[0][0] == '+' or user_timezone[0][0] == '-' \
+        else int(user_timezone[0][0:])
+    minutes = int(user_timezone[1])
+    if user_timezone[0][0] == '-':
+        return (datetime.now(timezone.utc) - timedelta(hours=hours, minutes=minutes)).strftime('%Y-%m-%d %H:%M:%S')
+    elif user_timezone[0][0] == '+':
+        return (datetime.now(timezone.utc) + timedelta(hours=hours, minutes=minutes)).strftime('%Y-%m-%d %H:%M:%S')
+    else:
+        return (datetime.now(timezone.utc)).strftime('%Y-%m-%d %H:%M:%S')
```

Comment:
why have you decided to handle signs manualy?
if you put '+1' to int(), you'll receive -- 1
if you put '-1' to int(), you'll receive --  -1
so what's the point?
you can pass negative values to timedelta(hours=

As well, I don't understand what is the use case for else block.

Reply:
Besides, you have redundant parenthesis in **else: return**